File: test-suite-java/src/test/java/io/micronaut/el/test/eligible/RegistrationService.java
package io.micronaut.el.test.eligible;

import io.micronaut.el.example.eligible.Eligible;
import jakarta.inject.Singleton;

@Singleton
public class RegistrationService {

    @Eligible(value = "#{ fn:adult(customer.age) && fn:inEurope(customer.country) }",
              otherwise = "#{ customer.name += ' must be an adult in Europe' }",
              name = "REGISTER")
    public String register(Customer customer) {
        return "registered " + customer.name();
    }

    @Eligible(value = "${ amount > 0 && customer.country == Locale.GERMANY.country }")
    public String deposit(Customer customer, long amount) {
        return "deposited " + amount + " for " + customer.name();
    }
}

File: doc-examples/example-kotlin/src/test/kotlin/example/RegistrationService.kt
package example

import io.micronaut.el.example.eligible.Eligible
import io.micronaut.el.example.eligible.MinAmount
import jakarta.inject.Singleton

@Singleton
open class RegistrationService {

    @Eligible(value = "#{ fn:adult(customer.age) && fn:inEurope(customer.country) }", // <1>
              otherwise = "#{ customer.name += ' must be an adult in Europe' }", // <2>
              name = "REGISTER") // <3>
    open fun register(customer: Customer): String = "registered " + customer.name

    @Eligible("\${ customer.country == Locale.GERMANY.country }") // <4>
    open fun deposit(customer: Customer,
                     @MinAmount(value = 100, inclusive = true,
                                message = "Must be greater than \${inclusive == true ? 'or equal to ' : ''}{value}") // <5>
                     amount: Long): String = "deposited $amount for " + customer.name
}
